### DISCUSSION TITLE
Best-practices for managing the json_key of supply

### DISCUSSION THREAD

**Discussion by Aw79** [fastlane/fastlane#18298] [Q&A]
Hi,

what are your best-practices for managing the `json_key` used by `supply`?

How do you store the json file, containing the Google service account?
To me it has a similar criticality like the signing credentials managed by `match`. With `match` the signing certificates (+ private key) and the provision profiles are stored encrypted in a repository. Although this is not about signing, but this json file is sensible for the interaction with Google PlayStore API.

Looking forward to learn how you work with `supply`.

Cheer, Andy 🤙

**Comment by dinsen** [comment]
Hi @Aw79 
You are not the only one thinking about this. See this:
https://github.com/fastlane/fastlane/discussions/18230

**Comment by Donald936** [comment]
2.) Purchase in volume for cost reduction. 
3.) Use GPO purchase contracts. 
4.) Establish par level inventory maintenance. 
5.) Obtain JIT Delivery Agreements. 
6.) Use technology where needed. 
7.) Manage risk and compliance.
